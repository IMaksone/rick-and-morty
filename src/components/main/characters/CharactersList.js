import { CharactersListEl } from "./CharactersListEl";
import { charactersBoxParams } from "../../../constatnts";

export const CharactersList = ({ listRef, render, length }) => {
  return (
    <div
      className="characters-list"
      ref={listRef}
      style={{
        height: charactersBoxParams.elHeight * length,
      }}
    >
      {render.map((ch, i) => (
        <CharactersListEl key={i} character={ch} />
      ))}
    </div>
  );
};
