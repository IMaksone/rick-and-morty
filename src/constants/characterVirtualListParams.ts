export const CHARACTER_VIRTUAL_LIST_PARAMS = {
  elHeight: 150,
  elFullHeight: 160
};

export type CharacterVirtualListParamsType = typeof CHARACTER_VIRTUAL_LIST_PARAMS;

export default CHARACTER_VIRTUAL_LIST_PARAMS;
