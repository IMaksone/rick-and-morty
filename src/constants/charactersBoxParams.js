export const charactersBoxParams = {
  elHeight: 150,
  elFullHeight: 160,
};
