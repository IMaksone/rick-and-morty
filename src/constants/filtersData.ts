import { FiltersData } from "src/types/filters";

export const initialFiltersData: FiltersData = {
  name: {
    type: "input",
    value: "",
  },
  type: { type: "input", value: "" },
  status: {
    type: "checkbox",
    valuesObject: { Alive: false, unknown: false, Dead: false },
  },
  species: {
    type: "checkbox",
    valuesObject: {
      Human: false,
      Alien: false,
      Humanoid: false,
      unknown: false,
      Poopybutthole: false,
      "Mythological Creature": false,
      Animal: false,
      Robot: false,
      Cronenberg: false,
      Disease: false,
    },
  },
  gender: {
    type: "checkbox",
    valuesObject: {
      Male: false,
      Female: false,
      unknown: false,
      Genderless: false,
    },
  },
};
