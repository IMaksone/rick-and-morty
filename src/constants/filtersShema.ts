type FilterTypesType = "input" | "checkbox";

export type FilterKeyType = "name" | "type" | "status" | "species" | "gender";

export type FilterType = {
  type: FilterTypesType;
  value?: string;
  valuesObject?: Object;
};

export type FilterCircuitType = {
  [key in FilterKeyType]?: FilterType;
};

export const FILTER_CIRCUIT: FilterCircuitType = {
  name: {
    type: "input",
    value: ""
  },
  type: { type: "input", value: "" },
  status: { type: "checkbox", valuesObject: {} },
  species: { type: "checkbox", valuesObject: {} },
  gender: { type: "checkbox", valuesObject: {} }
};
