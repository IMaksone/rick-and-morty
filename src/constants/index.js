export * from './filtersShema'
export * from './charactersBoxParams'
export * from './charactersFields'
export * from './characterSample'