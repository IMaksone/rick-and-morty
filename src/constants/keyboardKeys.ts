export const ESC_KEY = "Esc"