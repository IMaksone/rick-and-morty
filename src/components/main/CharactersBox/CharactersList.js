import PropTypes from "prop-types";

import { CharactersListEl } from "./CharactersListEl";
import { charactersBoxParams } from "../../../constants";
import { myCharacterType } from "../../../types/character";

export const CharactersList = ({ render, length }) => {
  return (
    <div
      className="characters-list"
      style={{
        height: charactersBoxParams.elHeight * length,
      }}
    >
      {render.map((ch, i) => (
        <CharactersListEl key={i} character={ch} />
      ))}
    </div>
  );
};

CharactersList.propTypes = {
  render: PropTypes.arrayOf(myCharacterType).isRequired,
  length: PropTypes.number.isRequired,
};

