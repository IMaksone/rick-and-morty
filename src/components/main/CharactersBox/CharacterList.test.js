import renderer from "react-test-renderer";

import { CharactersList } from "./CharactersList";
import { сharacterSample } from "../../../constants";

it("TEST CharactersList", () => {
  const component = renderer.create(
    <CharactersList render={[{ index: 0, data: сharacterSample }]} length={1} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
