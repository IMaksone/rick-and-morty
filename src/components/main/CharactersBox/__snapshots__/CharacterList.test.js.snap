// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST CharactersList 1`] = `
<div
  className="characters-list"
  style={
    Object {
      "height": 150,
    }
  }
>
  <div
    className="characters-list-el"
    style={
      Object {
        "height": 150,
        "top": 0,
      }
    }
  >
    <div
      className="character"
    >
      <img
        alt="character_image"
        className="character-img"
        onClick={[Function]}
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <div
        className="character-desc"
      >
        <div>
          <h3
            className="character-name text-light"
            onClick={[Function]}
          >
            Rick Sanchez
          </h3>
          <p
            className="character-text text-dark character-indicator character-indicator-alive"
          >
            alive
          </p>
        </div>
        <p
          className="character-text text-dark"
        >
          Human
        </p>
      </div>
    </div>
  </div>
</div>
`;
